pkg/chains/signing.go

```diff
@@ -44,6 +45,15 @@ type ObjectSigner struct {
 	Backends          map[string]storage.Backend
 	SecretPath        string
 	Pipelineclientset versioned.Interface
+	appContext        context.Context
```

Comment:
I don't think we want to attach contexts to the signer like this - this would prevent us from reusing the signer for multiple requests if there's a deadline attached to it. We already plumb through a context with Sign - we should reuse that if possible.

Reply:
The context here has a different purpose (which is probably a code smell).

The SBOM formater needs access to the OCI registry so it can read the SBOM (consider a private repo). To do this, we need a kube client to initialize the k8schain.

One way to get a kube client is to get it from the context, e.g. `kubeclient.Get(ctx)`. This, however, does not work with the context given to a reconcile iteration - the one `ObjectSigner.Sign` gets. It only works for the "application" context - the one `NewController` gets.

Instead of storing the application context, we could store the kube client directly in the ObjectSigner. That might make things more obvious.

@wlynch, wdyt?